### ISSUE TITLE
Curriculum learning 

### ISSUE THREAD

**Issue by BaiXue333** [lordmartian/deep_avsr#12]
You have the following description in Important Training Details：
"We perform iterations of Curriculum Learning by changing the PRETRAIN_NUM_WORDS config option. The number of words used in each iteration of curriculum learning is as follows: 1,2,3,5,7,9,13,17,21,29,37, i.e., 11 iterations in total."

I am confused. Can you describe in detail what the "iteration" refers to? My understanding is that the network need to iterate 1000 times ("NUM_STEPS")  every time "PRETRAIN_NUM_WORDS" is changed in the course learning strategy.

**Comment by lordmartian** [comment]
In the first iteration, you train the model with `PRETRAIN_NUM_WORDS=1`. This trained model is then used as a starting point for the second iteration where you train with `PRETRAIN_NUM_WORDS=2` and so on. 

3rd point in 'Important Training Details':

> In each iteration, the training is terminated forcefully once the validation set WER flattens. We also make sure that the Learning Rate has decreased to the minimum value before terminating the training.

You will need to terminate the training with 'Ctrl-C' once the validation WER stops improving and not carry on upto 1000 steps.

**Comment by BaiXue333** [comment]
Thank you very much for your reply. You mean that I  should manually execute the early stop and then pick up the best weights based on validation set WER from the checkpoints/models directory  as the initialization for the next iteration？
Can I understand this way？

**Comment by lordmartian** [comment]
Yes.
